src/main/python/ayab/signal_receiver.py

```diff
@@ -55,25 +59,19 @@ class SignalReceiver(QObject):
     hw_test_starter = Signal(Control)
     hw_test_writer = Signal(str)
 
-    def __init__(self):
+    def __init__(self)->None:
         super().__init__()
 
-    def signals(self):
-        """Iterator over names of signals."""
-        return filter(
-            lambda x: type(getattr(self, x)).__name__ == "SignalInstance",
-            SignalReceiver.__dict__.keys())
-
```

Comment:
I dep'd this function in favor of explicit definitions of the `SignalSender` methods, as it will allow more clear typing.